packages/split-button/src/SplitButton/SplitButton.tsx

```diff
@@ -56,7 +54,7 @@ export const SplitButton = InferredPolymorphic<SplitButtonProps, 'button'>(
     ref: React.Ref<any>,
   ) => {
     const { Component } = useInferredPolymorphic(as, rest, 'button');
-    const { darkMode, theme } = useDarkMode(darkModeProp);
+    const { theme } = useDarkMode(darkMode);
```

Comment:
We still want to use the `darkMode` value returned from `useDarkMode` because it reads and returns the `darkMode` value from the parent [LeafyGreenProvider](https://github.com/mongodb/leafygreen-ui/blob/main/packages/leafygreen-provider/src/DarkModeContext.tsx#L23-L28). If `darkMode` is passed into the component, it will take precedence over the `LeafyGreenProvider` `darkMode` value.

Reply:
> If `darkMode` is passed into the component, it will take precedence over the `LeafyGreenProvider` `darkMode` value.

That's the exact the behaviour I'd expect though?

I expect the `Button` and `Menu` (rendered internally in `SplitButton`) to call `useDarkMode` by them selves to apply the right theme from the provider iff a `darkMode` prop isn't passed to the `SplitButton`.

Reply:
Thank you for pointing this out! You're absolutely correct.

However, we’ll stick with the current implementation, as we intentionally pass `darkMode` directly to the `LeafyGreenProvider`. This approach is deliberate and aligns with how we typically handle `darkMode` across other LeafyGreen components.